vmdb/app/models/metric/ci_mixin.rb

```diff
@@ -107,12 +107,14 @@ def performances_maintains_value_for_duration?(options)
     window_starting_on = starting_on - duration
     $log.info("#{log_header} Reading performance records from: #{window_starting_on} to: #{now}")
 
-    select = "capture_interval_name, capture_interval, timestamp, #{column}" if Metric.column_names.include?(column.to_s)
-    total_records = self.send(meth).all(
-      :select     => select,
-      :conditions => ["capture_interval_name = ? and timestamp >= ? and timestamp < ?", interval_name, window_starting_on, now],
-      :order      => "timestamp DESC"
-    )
+    scope = send(meth)
+    if Metric.column_names.include?(column.to_s)
+      scope = scope.select "capture_interval_name, capture_interval, timestamp, #{column}"
+    end
+
+    total_records = scope.
+      where(["capture_interval_name = ? and timestamp >= ? and timestamp < ?", interval_name, window_starting_on, now]).
```

Comment:
The `capture_interval_name =` condition can be separated into a `.where`, right?